### PR TITLE
Only change parameter value when it is not set and fix error in metadata and set value to ShopID variable before using it

### PR DESCRIPTION
**Pull request by stasiukaitis-saulius**

### CODE REVIEW

File: core/fcpayone_events.php [hunk 1]
```diff
@@ -400,6 +400,8 @@ public static function addPayonePayments()
     {
         $oDb = oxDb::getDb();
 
+        $sShopId = $oDb->getOne("SELECT OXID FROM oxshops ORDER BY OXID ASC LIMIT 1");
```
Comment (andrefatchip) [inline core/fcpayone_events.php]: Will this give us the ShopId of the (Sub-)Shop in which the module currently has been activated? I doubt it. So my recommendation then would be the usage o the helper instance instead:

$sShopId = self::$_oFcpoHelper->fcpoGetConfig()->getShopId();

File: extend/application/models/fcPayOneBasketitem.php [hunk 2]
```diff
@@ -65,7 +65,7 @@ public function getArticle( $blCheckProduct = null, $sProductId = null, $blDisab
         // Leave value unchanged if it is explicitly forced from a usage.
         if (is_null($blCheckProduct) && $blSuccess && $sRefNr) {
             $blCheckProduct = !($blReduceStockBefore && $blSuccess && $sRefNr);
-        } else {
+        } elseif(is_null($blCheckProduct)) {
```
Comment (andrefatchip) [inline extend/application/models/fcPayOneBasketitem.php]: Makes sense.

File: metadata.php [hunk 3]
```diff
@@ -241,6 +241,6 @@
                 'block' => 'admin_roles_bemain_form',
                 'file' => 'fcpo_admin_roles_bemain_form',
         );
-        $aModule['extend'][] = array('roles_bemain' => 'fc/fcpayone/extend/application/controllers/admin/fcPayOneRolesBeMain');
+        $aModule['extend']['roles_bemain'] = 'fc/fcpayone/extend/application/controllers/admin/fcPayOneRolesBeMain';
```
Comment (andrefatchip) [inline metadata.php]: Confirmed.

File: core/fcpayone_events.php (hunk 1 above)
Comment (stasiukaitis-saulius) [inline core/fcpayone_events.php]: I guess so, but I am not 100% sure as I do not know what returns fcpoGetConfig.
Shop Config object would return active Shop id through a method getShopId.
Old usage could be something like: \oxRegistry::getConfig()->getShopId();

Comment (andrefatchip) [inline core/fcpayone_events.php]: fcpohelper is a class that extends oxBase and contains a simple wrapper method:

`    
     /**
     * Getter for config instance
     * 
     * @param void
     * @return \OxidEsales\Eshop\Core\Config
     */
    public function fcpoGetConfig() 
    {
        return $this->getConfig();
    }
`

However, I do NOT insist on this. If oxRegistry::getConfig()->getShopId() will do the job, I'll be absolutely fine with it :-)

**Pull request merged**